dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLMemoryStore.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.SemanticKernel.Http;
+using Microsoft.SemanticKernel.Memory;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureCosmosDBNoSQL;
+
+/// <summary>
+/// An implementation of <see cref="IMemoryStore"/> backed by a Azure Cosmos DB database.
+/// Get more details about Azure Cosmos DB vector search  https://learn.microsoft.com/en-us/azure/cosmos-db/
+/// </summary>
+public class AzureCosmosDBNoSQLMemoryStore : IMemoryStore, IDisposable
+{
+    private readonly CosmosClient _cosmosClient;
+    private readonly VectorEmbeddingPolicy _vectorEmbeddingPolicy;
+    private readonly IndexingPolicy _indexingPolicy;
+    private readonly string _databaseName;
+
+    /// <summary>
+    /// Initiates a AzureCosmosDBNoSQLMemoryStore instance using a Azure Cosmos DB connection string
+    /// and other properties required for vector search.
+    /// </summary>
+    /// <param name="connectionString">Connection string required to connect to Azure Cosmos DB.</param>
+    /// <param name="databaseName">The database name to connect to.</param>
+    /// <param name="vectorEmbeddingPolicy">The <see cref="VectorEmbeddingPolicy" /> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    /// <param name="indexingPolicy">The <see cref="IndexingPolicy"/> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    /// <param name="applicationName">The application name to use in requests.</param>
+    public AzureCosmosDBNoSQLMemoryStore(
+        string connectionString,
+        string databaseName,
+        VectorEmbeddingPolicy vectorEmbeddingPolicy,
+        IndexingPolicy indexingPolicy,
+        string? applicationName = null)
+    {
+        var options = new CosmosClientOptions
+        {
+            ApplicationName = applicationName ?? HttpHeaderConstant.Values.UserAgent,
+            Serializer = new CosmosSystemTextJsonSerializer(JsonSerializerOptions.Default),
+        };
+        this._cosmosClient = new CosmosClient(connectionString, options);
+        this._databaseName = databaseName;
+        this._vectorEmbeddingPolicy = vectorEmbeddingPolicy;
+        this._indexingPolicy = indexingPolicy;
```

Comment:
In the latest change I added a check for the the `vectorEmbeddingPolicy`.  I didn't for the `indexingPolicy` since it can technically work, Cosmos DB will just end up doing a brute force search that will be slow and expensive. Maybe we should be more prescriptive than that though?

Reply:
That sounds like the right balance to me. I had assumed it was required in indexingPolicy as well for vector search to work, but if it's not required by Cosmos DB, I don't think we should make it required in our sdk.